mqtt-impl/src/main/java/io/streamnative/pulsar/handlers/mqtt/support/DefaultProtocolMethodProcessorImpl.java

```diff
@@ -459,11 +465,21 @@ public void processUnSubscribe(Channel channel, MqttUnsubscribeMessage msg) {
         FutureUtil.waitForAll(futureList).thenAccept(__ -> {
             // ack the client
             int messageID = msg.variableHeader().messageId();
+            int protocolVersion = NettyUtils.getProtocolVersion(channel);
             MqttFixedHeader fixedHeader = new MqttFixedHeader(MqttMessageType.UNSUBACK, false, MqttQoS.AT_MOST_ONCE,
                     false, 0);
-            MqttUnsubAckMessage ackMessage = new MqttUnsubAckMessage(fixedHeader,
-                    MqttMessageIdVariableHeader.from(messageID));
-
+            MqttMessage ackMessage;
+            // support mqtt5
+            if (protocolVersion == MqttVersion.MQTT_5.protocolLevel()) {
+                MqttMessageIdAndPropertiesVariableHeader mqttMessageIdAndPropertiesVariableHeader =
+                        new MqttMessageIdAndPropertiesVariableHeader(messageID, MqttProperties.NO_PROPERTIES);
+                MqttUnsubAckPayload mqttUnsubAckPayload =
+                        new MqttUnsubAckPayload(MqttConnectReturnCode.CONNECTION_ACCEPTED.byteValue());
```

Comment:
We'd better define a new unsuback code instead of using Netty connections.

Reply:
Ok, i will fix it ASAP